### PR TITLE
Feature/oauth2 kalila

### PR DESCRIPTION
**Pull request by kasenvr**

### CODE REVIEW

File: interface/resources/qml/LoginDialog/LinkAccountBody.qml [hunk 1]
```diff
@@ -436,7 +469,7 @@ Item {
                 font.pixelSize: linkAccountBody.textFieldFontSize
                 font.bold: linkAccountBody.fontBold
 
-                text: "<a href='metaverse.vircadia.com/users/password/new'> Can't access your account?</a>"
+                text: "<a href='http://metaverse.vircadia.com/users/password/new'> Can't access your account?</a>"
```
Comment (ctrlaltdavid) [inline interface/resources/qml/LoginDialog/LinkAccountBody.qml:472]:
```suggestion
                text: "<a href='https://metaverse.vircadia.com/users/password/new'> Can't access your account?</a>"
```

File: interface/resources/qml/LoginDialog/LinkAccountBody.qml [hunk 2]
```diff
@@ -562,7 +595,7 @@ Item {
                      leftMargin: hifi.dimensions.contentSpacing.x
                 }
 
-                text: "<a href='metaverse.vircadia.com/users/register'>Sign Up</a>"
+                text: "<a href='http://metaverse.vircadia.com/users/register'>Sign Up</a>"
```
Comment (ctrlaltdavid) [inline interface/resources/qml/LoginDialog/LinkAccountBody.qml:598]:
```suggestion
                text: "<a href='https://metaverse.vircadia.com/users/register'>Sign Up</a>"
```

File: interface/src/Menu.cpp [hunk 3]
```diff
@@ -9,6 +9,10 @@
 //  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
 //
 
+// For happ(ier) development of QML, use these two things:
+// This forces QML files to be pulled from the source as you edit it: HIFI_USE_SOURCE_TREE_RESOURCES=1
```
Comment (ctrlaltdavid) [inline interface/src/Menu.cpp:13]:
```suggestion
// This forces QML files to be pulled from the source as you edit it: set environment variable HIFI_USE_SOURCE_TREE_RESOURCES=1
```

File: interface/src/ui/DialogsManager.cpp [hunk 4]
```diff
@@ -110,10 +110,29 @@ void DialogsManager::setDomainConnectionFailureVisibility(bool visible) {
     }
 }
 
+void DialogsManager::requestMetaverseLoginState() {
+    DialogsManager::setMetaverseLoginState();
+}
+
+void DialogsManager::requestDomainLoginState() {
+    DialogsManager::setDomainLoginState();
+}
```
Comment (ctrlaltdavid) [inline interface/src/ui/DialogsManager.cpp:119]: Why these synonyms for the "set" methods? (Is there a "TODO" here that should be marked?)

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR589/Vircadia-Alpha-PR589-cb0371c_448cdcb3.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR589/Vircadia-Alpha-PR589-cb0371c_448cdcb3.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR589/Vircadia-Alpha-PR589-cb0371c.tar_448cdcb3.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR589/Vircadia-Alpha-PR589-cb0371c.tar_448cdcb3.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR589/Vircadia-Alpha-PR589-cb0371c_448cdcb3.exe

File: interface/src/ui/DialogsManager.cpp (hunk 4 above)
Comment (kasenvr) [reply interface/src/ui/DialogsManager.cpp:119]: There might have been, but I cannot remember now. We'll find out when we need it again. :)

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR589/Vircadia-Alpha-PR589-d317650_d89cf867.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR589/Vircadia-Alpha-PR589-d317650_d89cf867.exe

**Pull request merged**